### PR TITLE
Package eliom.11.0.0

### PR DESCRIPTION
**Pull request by balat**
### `eliom.11.0.0`
Advanced client/server Web and mobile framework
Eliom is a framework for implementing Web sites and client/server Web and mobile applications.
It uses advanced concepts to simplify the implementation of common behaviors (e.g. scoped sessions, continuation based Web programming ...).
It uses advanced static typing features of OCaml to check many properties of the Web application at compile-time (html, page parameters ...).
Eliom allows implementing the whole application as a single program that includes both the client and the server code. For example, you can implement event handlers (onclick ...) directly in OCaml, and you can call a server-side OCaml function from the client. Pages are generated either on the server or the client.
These client-side features remain compatible with traditional Web programming (links, forms, URLs, bookmarks, sessions ...).
It is possible to generate mobile applications for Android and iOS with the exact same code as your Web application.
The client-side code is compiled to JS using Ocsigen Js_of_ocaml or to Wasm using Wasm_of_ocaml.

---
* Homepage: https://ocsigen.org/eliom/
* Source repo: git+https://github.com/ocsigen/eliom.git
* Bug tracker: https://github.com/ocsigen/eliom/issues

---
:camel: Pull-request generated by opam-publish v2.3.0

### CODE REVIEW

**Comment by balat** [comment]
Needs Ocsigen Server 6. See #26449

File: packages/eliom/eliom.11.0.0/opam [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+The client-side code is compiled to JS using Ocsigen Js_of_ocaml or to Wasm using Wasm_of_ocaml.
+"""
+maintainer: ["dev@ocsigen.org"]
+authors: ["dev@ocsigen.org"]
+license: "LGPL-2.1-only WITH OCaml-LGPL-linking-exception"
+homepage: "https://ocsigen.org/eliom/"
+bug-reports: "https://github.com/ocsigen/eliom/issues"
+depends: [
+  "dune" {>= "3.6"}
+  "ocaml" {>= "4.08.0"}
```
Comment (shonfeder) [inline packages/eliom/eliom.11.0.0/opam:20]: CI is current failing on ocaml versions =< 4.11 due to the injectivity annotation, which apparently was added in 4.12 (e.g., in https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/8a7c2e44a912ad011644f5ad1a9808ceb3a3425b/variant/compilers,4.08,eliom.11.0.0):

```
#=== ERROR while compiling eliom.11.0.0 =======================================#
# context              2.3.0~alpha~dev | linux/x86_64 | ocaml-base-compiler.4.11.2 | pinned(https://github.com/ocsigen/eliom/archive/refs/tags/11.0.0.tar.gz)
# path                 ~/.opam/4.11/.opam-switch/build/eliom.11.0.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p eliom -j 31 @install
# exit-code            1
# env-file             ~/.opam/log/eliom-7-505e61.env
# output-file          ~/.opam/log/eliom-7-505e61.out
### output ###
# (cd _build/.sandbox/5e828d3fdeceadb53ee04631f1c5620d/default && .ppx/13d6111584a1a9a5b7ca130081cf0d43/ppx.exe --cookie 'library-name="eliom_server"' -o src/lib/server/eliommod.pp.mli --intf src/lib/server/eliommod.mli -corrected-suffix .ppx-corrected -diff-cmd - -dump-ast)
# File "src/lib/server/eliommod.mli", line 25, characters 7-8:
# 25 |   type !'a t
#             ^
# Error: Syntax error: 'end' expected
# File "src/lib/server/eliommod.mli", line 23, characters 11-14:
# 23 | module S : sig
#                 ^^^
#   This 'sig' might be unmatched



<><> Error report <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+- The following actions failed
| - build eliom 11.0.0
```

So I think you'll either need to drop that annotation or lift the lower bound here:

```suggestion
  "ocaml" {>= "4.12"}
```

**Review by shonfeder — COMMENTED** [review]
There is also a lower bounds error being triggered in # File "src/

```
lib/server/eliommod.mli", line 94, characters 5-24:
# 94 |   -> Xml_light_types.xml
#           ^^^^^^^^^^^^^^^^^^^
# Error: Unbound module Xml_light_types



<><> Error report <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+- The following actions failed
| - build eliom 11.0.0
+-
```

https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/8a7c2e44a912ad011644f5ad1a9808ceb3a3425b/variant/compilers,4.12,eliom.11.0.0,lower-bounds

I haven't yet determined exactly what bound needs to be raised tho, since `xml-light` seems to have always had this module afaict.

File: packages/eliom/eliom.11.0.0/opam [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  "lwt_log"
+  "lwt_ppx" {>= "1.2.3"}
+  "tyxml" {>= "4.6.0" & < "4.7.0"}
+  "ocsigenserver" {>= "6.0.0" & < "7.0.0"}
+  "ipaddr" {>= "2.1"}
+  "reactiveData" {>= "0.2.1"}
+  "base-bytes"
+  "ocsipersist" {>= "2.0" & < "3.0"}
+  "ppx_optcomp"
+  "odoc" {with-doc}
```
Comment (shonfeder) [inline packages/eliom/eliom.11.0.0/opam:40]: Ah, I am mistaken about the API of xml-light, seems that the needed module type was actually introduced in 2.5, so I think we either need

```suggestion
  "odoc" {with-doc}
  "xml-light" {>= "2.5"}
```

here or else in `ocisgenserver` (from which the xml-light dependency is being being picked up transitively).

Comment (balat) [reply packages/eliom/eliom.11.0.0/opam:40]: Thank you! I will add these constraints

**Comment by shonfeder** [comment]
The windows failure is due to the ssl package, and the only remaining ocaml-ci failures are solver timeouts. All looks good here.

Thanks for publishing the update!

**Review by shonfeder — APPROVED** [review]

**Comment by balat** [comment]
Thanks!

**Pull request merged**